pvlib/pvsystem.py

```diff
@@ -2537,6 +2553,73 @@ def pvwatts_losses(soiling=2, shading=3, snow=0, mismatch=2, wiring=2,
     return losses
 
 
+def irradiance_loss_pvsyst(effective_irradiance, irradiance_shading_loss,
+                           irradiance_snow_loss, irradiance_soiling_loss):
+    r"""
+    Calculates irradiance losses using a model based on PVsyst.
+
+    The PVsyst method [1]_ linearly reduces the effective irradiance to
+    account for the effects of shading, snow, and soiling. The separate
+    loss factors are compounded to calculate a single loss factor.
+
+    The separate losses are compounded using the following equation:
+
+    .. math::
+
+        L_{total}(\%) = 100 [ 1 - \Pi_i ( 1 - \frac{L_i}{100} ) ]
+
+    :math:L_{total} is the total irradiance loss resulting from the effects of
+    shading, snow, and soiling.
+
+    Note the parameters must each be a series with a DatetimeIndex.  The
+    index of the returned series will be the same as the effective_irradiance
+    index.  All other parameters will be resampled to match this index with
+    a "fill forward" method for filling holes.
```

Comment:
Why do we do this in this function? It's non-standard behavior in pvlib. 

Perhaps it's discussed in the PR comments above (which I have not read) but if it's important it should be in the documentation or in a comment.

Reply:
Reading the test code I now understand the motivation. I'm not convinced that it's the right approach for a general purpose library. Options are:

1. we do nothing and let numpy/pandas broadcasting/indexing do its thing
2. we expose a `fill_method` argument or keyword argument in the function
3. leave it as is
4. something else?

Reply:
I believe exposing a fill_method argument makes sense.  The reason for an explicit outline of the broadcasting method is to ease the transition of users coming from PVSyst.  There, you input a soiling loss value for each month.  Ideally, a user could input an array of 12 values for the soiling_loss parameter, and the function would be able to broadcast them to the appropriate month.  However, I wanted this to maintain flexibility for multi-year simulations with different monthly values and ultimately settled on the solution that I did.  I'm always open to other ideas, but I think we will want to make the results of broadcasting obvious.  (Maybe that's just in a tutorial.)